Protocol.md

```diff
@@ -1101,6 +1102,33 @@ key pair and the other client's session key pair.
 }
 ```
 
+## 'application' Message
+
+Once the client-to-client handshake has been completed, the user
+application of a client MAY trigger sending this message.  
+This message allows user applications to send simple control
+messages/early data or whatever it desires without having to modify an
+existing task. However, this message SHOULD NOT be abused to write
+custom protocols. Tasks SHOULD support this message to be sent at any
+time.
```

Comment:
Depends on the implementation :grin:. There might be use cases where the task cannot support an 'application' message. In terms of implementation design, I'd make the client instance responsible, but the task should be able to turn it off and change the underlying transport for handover.